kernel/src/process.rs

```diff
@@ -790,7 +791,7 @@ pub trait Process {
     ///
     /// Returns `true` if the upcall function pointer is valid for this process,
     /// and `false` otherwise.
-    fn is_valid_upcall_function_pointer(&self, upcall_fn: NonNull<()>) -> bool;
```

Comment:
Is this another change that would later be reverted once the TODO in https://github.com/tock/tock/pull/4174/files#diff-dc3a689288cb588e11aee7e8fc2b3e8d3a361c4111769e1b8e0df862f5ba693fR879 is dealt with?

That becomes a bit harder to track across multiple files... maybe can try to do something like contain all of the temporary `NonNull` removals to one commit, and then add to the comment linked above a reference to the commit to revert when the TODO resolves?

Reply:
Left a comment linking to the CHERI tracker.